src/Randomizer.SMZ3.Tracking/AutoTracking/GameService.cs

```diff
@@ -124,6 +124,221 @@ public bool TryHealPlayer()
             return false;
         }
 
+        /// <summary>
+        /// Fully fills the player's magic
+        /// </summary>
+        /// <returns>False if it is currently unable to give magic to the player</returns>
+        public bool TryFillMagic()
+        {
+            if (_autoTracker == null || !_autoTracker.IsConnected || _autoTracker.CurrentGame != Game.Zelda)
+            {
+                return false;
```

Comment:
This is just personal preference, but I'd throw here, and instead check for various connections at the caller, so Tracker can give more accurate responses, e.g. "Maybe you should start playing the game first" or "I don't think Samus can use magic". 

I could probably do this myself later if you want as part of the refactoring.

Reply:
So if you pull up the CheatsModule, I'm actually already doing this. The PlayerCanCheat function verifies that the player has cheats enabled as well as confirming that auto tracking is enabled and connected. So for the CheatsModule the only thing that should fail is if not in the game or something along those lines. I just kept this in here as a fail safe.

If you'd prefer me to use exception catching rather than the try returning bool paradigm, I can make that change though.

Reply:
I've just been reviewing the PR changes one at a time, so I probably looked past that.

Considering these cases are covered elsewhere, and especially _because_ it may be used elsewhere, I personally feel throwing exceptions makes more sense than using `TryX`. This is really rather minor though, so don't worry about making that change.